### DISCUSSION TITLE
query multiple ttl files with sparql

### DISCUSSION THREAD

**Discussion by luisenriqueramos1977** [eclipse-rdf4j/rdf4j#4768] [General]
I have two simple ttl files with following data:

example1.ttl:
ns1:place1 a owl:NamedIndividual;
                 ns2:hasAddress address_1 .

example2.ttl:
ns2:address_1 a owl:NamedIndividual;
                 ns2:hasproperty1 property1 .
                 ns2:hasproperty2 property2 .

I uploaded the files with:

con.add(example1.ttl)
g.add(example2.ttl)

when I use sparql to get the address of place:

SELECT ?address
WHERE {  
        ns1:place1  ns2:hasAddress ?address .
        }

I get the expected address object

when I ask for all properties of address object:
SELECT ?property ?value 
WHERE {  
        ns2:address_1 ?property ?value .
        }
I also get them, but when I ask for the properties of the address of the place:

SELECT  ?property ?value
WHERE {  
        ns1:place1  ns2:hasAddress ?address .
        ?address ?property ?value .
        }

I get a blank result, I wonder what should I be doing wrong?, 

Because I was waiting to get all properties of the address.

Best regards

Luis

**Comment by hmottestad** [comment]
Can you post your Java code?

**Reply by luisenriqueramos1977** [reply]
yes sure:

```
File dataDir = new File("a_path");
Repository repo = new SailRepository(new MemoryStore(dataDir));	
//adding data to repository
File example1_rdf = new File("example1.ttl");
File example2_rdf  = new File("example2.ttl");

try {
RepositoryConnection con = repo.getConnection();
try {
con.add(example1_rdf,RDFFormat.TURTLE);
//con.add(example1_rdf, RDFFormat.TURTLE);
 }
finally {
	con.close();
}
	}
	catch (RDF4JException e) {
	// handle exception
}

///query
RepositoryConnection con = repo.getConnection();


query1="SELECT ?address ?property ?value
WHERE {
ns1:place1 ns2:hasAddress ?address .
?address ?property ?value .
}"
String queryString = query1;
TupleQuery tupleQuery = con.prepareTupleQuery(queryString);

try (TupleQueryResult result = tupleQuery.evaluate()) {
						      while (result.hasNext()) {  // iterate over the result
						         BindingSet bindingSet = result.next();
						         Value subject1 = bindingSet.getValue("address");
						       System.out.println("the address: "+subject1);
						       
						      }
						   }

```



result is blank

**Reply by kenwenzel** [reply]
Where do you add the file `example2_rdf`?
